### ISSUE TITLE
Download translation & Weblate-specific data

### ISSUE THREAD

**Issue by marcperrin** [WeblateOrg/weblate#1194]
Hello,
When downloading for offline translation, you get the store data but you never get Weblate-specific data such as Suggestions, Weblate Comments, or Weblate source string tags like check-length etc.
Do you think it would be appropriate to get this data at download\* so that offline translator has a level of information closer to online translation?
Thanks!

*) I guess all formats have a comment zone where all of this could be fed. 
For .po, we could be a bit more specific, as Weblate comments and suggestions could qualify as translator comments (#) (rather than extracted comments #.), and for Weblate flags, maybe flags? (#,)
(Or maybe you'd envision that as a separate downloaded file??)
##

**Comment by nijel** [comment]
Something like this is definitely possible, though it really needs good specification to make it usable.

For example using flags for anything non standard in Gettext is quite tricky, many of the tools do silently strip non standard flags...

**Comment by marcperrin** [comment]
OK then, the less risky approach would be to keep everything as comments 
(e.g. # [Weblate data] { json data... })
You might also get problems with the "Merge translation comments" option at upload, but then again, I think there is already a problem with that option, as I tend to get duplicated comments if I keep this box checked.

**Comment by nijel** [comment]
Flags could be indeed exported to Xliff and gettext PO files (the latter is blocked by https://github.com/translate/translate/pull/3913 as they will otherwise be completely messed up). That's something I will definitely do for 3.9 release.

Comments could be turned into xliff [notes](http://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html#note) and gettext translator comments.

I'm not really sure about suggestions as those really do not translate well into anything existing.

**Comment by nijel** [comment]
Implementation for above outlined case is in https://github.com/WeblateOrg/weblate/pull/2962

**Comment by nijel** [comment]
Thank you for your report, the issue you have reported has just been fixed. 

* In case you see problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).
